modules/wallet/encrypt.go

```diff
@@ -193,29 +193,48 @@ func (w *Wallet) managedUnlock(masterKey crypto.TwofishKey) error {
 	subscribed := w.subscribed
 	w.mu.RUnlock()
 	if !subscribed {
-		// Subscription can take a while, so spawn a goroutine to print the
-		// wallet height every few seconds. (If subscription completes
-		// quickly, nothing will be printed.)
-		done := make(chan struct{})
-		go w.rescanMessage(done)
-		defer close(done)
-
-		err = w.cs.ConsensusSetSubscribe(w, lastChange)
-		if err == modules.ErrInvalidConsensusChangeID {
-			// something went wrong; resubscribe from the beginning
-			err = dbPutConsensusChangeID(w.dbTx, modules.ConsensusChangeBeginning)
+		if primarySeedProgress == 0 {
+			// If the wallet has not been used, we don't need to scan the entire
+			// blockchain looking for relevant transactions. Instead, just grab
+			// the current height and subscribe to future blocks.
+			err = w.cs.ConsensusSetSubscribe(w, modules.ConsensusChangeRecent)
 			if err != nil {
-				return fmt.Errorf("failed to reset db during rescan: %v", err)
+				return fmt.Errorf("wallet subscription failed: %v", err)
 			}
-			err = dbPutConsensusHeight(w.dbTx, 0)
+			w.mu.Lock()
+			err = dbPutConsensusHeight(w.dbTx, w.cs.Height())
+			w.mu.Unlock()
```

Comment:
this part feels janky. When a new block arrives, does `cs.Height` return the new height before or after updating subscribers? If it's after, this should be safe. It it's before, then this could result in an off-by-one in the height. Would it make more sense to set the height before subscribing?